web-console/src/dialogs/table-action-dialog.tsx

```diff
@@ -0,0 +1,231 @@
+import * as FileSaver from 'file-saver';
+import * as React from 'react';
+import * as CopyToClipboard from 'react-copy-to-clipboard';
+
+import {AppToaster} from '../singletons/toaster';
+import { getDruidErrorMessage } from '../utils';
+
+import './table-action-dialog.scss';
+
+interface TableActionDialogProps extends React.Props<any> {
+  onClose: () => void;
+  metaData: TableActionDialogMetaData;
+  terminateSupervisor: (id: string) => void;
+  resetSupervisor: (id: string) => void;
+  resumeSupervisor: (id: string) => void;
+  suspendSupervisor: (id: string) => void;
+  killTask: (id: string) => void;
+}
+
+interface TableActionDialogState {
+  jsonValue: string;
+  activeEndpoint: '' | 'status' | 'stats' | 'history' | 'reports' | 'log?offset=-8192' | 'log';
+}
+
+export interface TableActionDialogMetaData {
+  id: string | null;
+  mode: 'supervisor' | 'task' | null;
+  status: 'RUNNING' | 'WAITING' | 'PENDING' | 'SUCCESS' | 'FAILED' | null;
+  supervisorSuspended: boolean | null;
+}
+
+export class TableActionDialog extends React.Component<TableActionDialogProps, TableActionDialogState> {
+  constructor(props: TableActionDialogProps) {
+    super(props);
+    this.state = {
+      jsonValue: '',
+      activeEndpoint: ''
+    };
+  }
+
+  private getJsonInfo = async (endpoint: string) => {
+    const { mode, id } = this.props.metaData;
+    try {
+      const resp = await axios.get(`/druid/indexer/v1/${mode}/${id}/${endpoint}`);
+      const data = resp.data;
+      this.setState({
+        jsonValue: JSON.stringify(data, undefined, 2)
+      });
+    } catch (e) {
+      AppToaster.show({
+        message: e.response.data,
+        intent: Intent.WARNING
+      });
+      this.setState({
+        jsonValue: ''
+      });
+    }
+  }
+
+  private downloadJson = () => {
+    const { id, mode } = this.props.metaData;
+    const { jsonValue, activeEndpoint } = this.state;
+
+    const blob = new Blob([jsonValue], {
+      type: 'text/json'
+    });
+
+    FileSaver.saveAs(blob, `${mode}-${id}-${activeEndpoint === '' ? 'payload' : activeEndpoint}.json`);
+  }
+
+  render() {
+    const { onClose, terminateSupervisor, resetSupervisor, resumeSupervisor, suspendSupervisor } = this.props;
+    const { jsonValue, activeEndpoint } = this.state;
+    const { id, mode, status, supervisorSuspended } = this.props.metaData;
+
+    const supervisorTableSideButtonMetadata = [
+      {icon: 'align-left', text: 'Payload', endpoint: ``},
+      {icon: 'dashboard', text: 'Status', endpoint: `status`},
+      {icon: 'chart', text: 'Statistics', endpoint: `stats`},
+      {icon: 'history', text: 'History', endpoint: `history`}
+    ];
+
+    const taskTableSideButtonMetadata = [
+      {icon: 'align-left', text: 'Payload', endpoint: ``},
+      {icon: 'dashboard', text: 'Status', endpoint: `status`},
+      {icon: 'comparison', text: 'Reports', endpoint: `reports`},
+      {icon: 'align-justify', text: 'Logs', endpoint: `log?offset=-8192`}
+    ];
```

Comment:
I do not think it makes sense to merge SupervisorActionsDialog and TasksActionsDialog into one high level TableActionDialog. You should have a TableActionDialog that is like the snitch dialog (never directly used but used by a `SupervisorActionsDialog` and a `TasksActionsDialog`. The `TableActionDialog` should take care of the (grid) layout and the tabs, the specific dialogs should add their own views.